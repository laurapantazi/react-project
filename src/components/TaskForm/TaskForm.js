import React, {Component} from 'react'
import { Form, Button, Col, InputGroup } from 'react-bootstrap'
import { connect } from 'react-redux'
import { addTask } from '../../store/actions'

class TaskForm extends Component {
  constructor (props) {
    super(props)
    this.state = {
      description: '',
      category: '',
      duration: '',
      priority: '',
      points: '',
      status: false,
    }
  }

  handleInputChange = (event) => {
    const {value, name} = event.target
    this.setState({
      [name]: value
    })
  }

  handleSubmit = (event) => {
    event.preventDefault()
    let {description, category, duration, priority, points, status} = this.state
    this.props.addTask(description, category, duration, priority, points, status)
    this.setState({ description: '', category: '', duration: '', priority: '', points: ''})
  }

  render () {
    return (
      <div className="container taskList">
        <Form onSubmit={this.handleSubmit}>
          <Form.Group controlId="formPlaintextDescription">
            <Form.Label>Description</Form.Label>
            <Form.Control type="text" name="description" value={this.state.description} placeholder="Type description" onChange={this.handleInputChange} />
          </Form.Group>

          <Form.Group controlId="formPlaintextCategory">
            <Form.Label>Category</Form.Label>
            <Form.Control type="text" name="category" value={this.state.category} placeholder="Type category" onChange={this.handleInputChange} />
          </Form.Group>

          <Form.Row>
            <Form.Group as={Col} controlId="formGridDuration">
              <Form.Label>Duration</Form.Label>
              <InputGroup>
              <Form.Control type="number" name="duration" value={this.state.duration} placeholder="Type duration" onChange={this.handleInputChange} />
              <InputGroup.Append>
                <InputGroup.Text id="inputGroupPrepend">minutes</InputGroup.Text>
              </InputGroup.Append>
              </InputGroup>
            </Form.Group>

            <Form.Group as={Col} controlId="formGridPriority">
              <Form.Label>Priority</Form.Label>
              <Form.Control as="select" name="priority" value={this.state.priority} onChange={this.handleInputChange}>
                <option>Choose Priority</option>
                <option>High</option>
                <option>Medium</option>
                <option>Low</option>
              </Form.Control>
            </Form.Group>

            <Form.Group as={Col} controlId="formGridPoints">
              <Form.Label>Points</Form.Label>
              <Form.Control type="number" step="5" name="points" value={this.state.points} placeholder="Type points" onChange={this.handleInputChange}/>
            </Form.Group>
          </Form.Row>
          <Button variant="primary" type="submit">
            Add Task
          </Button>
        </Form>
      </div>
    )
  }
}

export default connect(null, { addTask })(TaskForm)