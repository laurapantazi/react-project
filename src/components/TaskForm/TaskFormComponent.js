import React, {Component} from 'react'
import { Form, Button, Col, InputGroup, FormControl, ListGroup } from 'react-bootstrap'
import PropTypes from 'prop-types'

export class TaskFormComponent extends Component {
  constructor (props) {
    super(props)
    this.state = {
      id: null,
      description: '',
      category: '',
      duration: '',
      priority: '',
      points: '',
      status: false,
      errors: []
    }
  }
  componentDidMount () {
    if (this.props.editTask) {
      this.setState({
        id: this.props.task.id,
        description: this.props.task.description,
        category: this.props.task.category,
        duration: this.props.task.duration,
        priority: this.props.task.priority,
        points: this.props.task.points,
        status: this.props.task.status,
      })
    }
  }
  handleInputChange = (event) => {
    const {value, name} = event.target
    this.setState({
      [name]: value
    })
  }

  formHasErrors = () => {
    this.setState({errors: []})
    let {description, category} = this.state
    let errors = []
    if (!description) {
      errors.push('Task description is required')
    }
    if (!category) {
      errors.push('Task category is required')
    }
    if (errors.length) {
      this.setState(previousState => ({
        errors: [...previousState.errors, ...errors]
      }))
      return true
    } else {
      return false
    }
  }

  handleSubmit = (event) => {
    event.preventDefault()
    let {id, description, category, duration, priority, points, status} = this.state
    if (!this.formHasErrors()) {
      if (this.props.editTask) {
        this.props.onSave(id, description, category, duration, priority, points, status)
      } else {
        this.props.onSave(description, category, duration, priority, points, status)
        this.setState({ description: '', category: '', duration: '', priority: '', points: ''})
      }
    }
  }

  showErrors () {
    return this.state.errors.map((error, index) => {
      return (
        <ListGroup.Item as="li" variant="danger" key={index}>
          {error}
        </ListGroup.Item>
      )
    })
  }
  render () {
    return (
      <div className="container taskList">
        <ListGroup as="ul">
          {this.showErrors()}
        </ListGroup>
        <Form onSubmit={this.handleSubmit}>
        {this.state.errors.description}
          <Form.Group controlId="formPlaintextDescription">
            <Form.Label>Description</Form.Label>
            <Form.Control type="text" name="description" value={this.state.description} placeholder="Type description" onChange={this.handleInputChange} />
            <FormControl.Feedback>
              {this.state.errors.description}
            </FormControl.Feedback>
          </Form.Group>

          <Form.Group controlId="formPlaintextCategory">
            <Form.Label>Category</Form.Label>
            <Form.Control type="text" name="category" value={this.state.category} placeholder="Type category" onChange={this.handleInputChange} />
          </Form.Group>

          <Form.Row>
            <Form.Group as={Col} controlId="formGridDuration">
              <Form.Label>Duration</Form.Label>
              <InputGroup>
              <Form.Control type="number" name="duration" value={this.state.duration} placeholder="Type duration" onChange={this.handleInputChange} />
              <InputGroup.Append>
                <InputGroup.Text id="inputGroupPrepend">minutes</InputGroup.Text>
              </InputGroup.Append>
              </InputGroup>
            </Form.Group>

            <Form.Group as={Col} controlId="formGridPriority">
              <Form.Label>Priority</Form.Label>
              <Form.Control as="select" name="priority" value={this.state.priority} onChange={this.handleInputChange}>
                <option>Choose Priority</option>
                <option>High</option>
                <option>Medium</option>
                <option>Low</option>
              </Form.Control>
            </Form.Group>

            <Form.Group as={Col} controlId="formGridPoints">
              <Form.Label>Points</Form.Label>
              <Form.Control type="number" step="5" name="points" value={this.state.points} placeholder="Type points" onChange={this.handleInputChange}/>
            </Form.Group>
          </Form.Row>
          <Button variant="primary" type="submit">
            Add Task
          </Button>
        </Form>
      </div>
    )
  }
}

TaskFormComponent.propTypes = {
  edit: PropTypes.bool,
  add: PropTypes.bool,
  task: PropTypes.object,
  editTask: PropTypes.func,
  addTask: PropTypes.func 
}