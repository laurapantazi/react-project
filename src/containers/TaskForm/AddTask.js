import React, { Component } from 'react'
import { connect } from 'react-redux'
import { addTask } from 'store/actions'
import {TaskForm} from 'components/TaskForm/TaskForm'

class AddTask extends Component {
  render () {
    return (
      <TaskForm add addTask={this.props.addTask} />
    )
  }
}
export default connect(null, { addTask })(AddTask)