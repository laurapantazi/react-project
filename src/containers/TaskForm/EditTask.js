/*eslint eqeqeq: "off"*/
import React, {Component} from 'react'
import { connect } from 'react-redux'
import { editTask } from 'store/actions'
import {TaskForm} from 'components/TaskForm/TaskForm'
import { Redirect } from 'react-router-dom'

class EditTask extends Component {
  constructor (props) {
    super(props)
    this.state = {id: this.props.match.params.id}
  }
  render () {
    let taskExists = this.props.tasks.find(task => task.id == this.state.id)
    if (taskExists) {
      return (
        <TaskForm edit task={taskExists} editTask={this.props.editTask} />
      )
    } else {
      return (<Redirect to="/404" />)
    }
  }
}

function mapStateToProps (state) {
  return {tasks: state.tasks}
}
export default connect(mapStateToProps, { editTask })(EditTask)
