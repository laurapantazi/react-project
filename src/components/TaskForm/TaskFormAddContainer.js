import React, { Component } from 'react'
import { connect } from 'react-redux'
import { addTask } from '../../store/actions'
import {TaskFormComponent} from './TaskFormComponent'

class TaskFormAddContainer extends Component {
  render () {
    return (
      <TaskFormComponent add addTask={this.props.addTask} />
    )
  }
}
export default connect(null, { addTask })(TaskFormAddContainer)