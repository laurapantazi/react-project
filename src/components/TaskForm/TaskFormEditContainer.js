import React, {Component} from 'react'
import { connect } from 'react-redux'
import { editTask } from '../../store/actions'
import {TaskFormComponent} from './TaskFormComponent'
import { Redirect } from 'react-router-dom'

class TaskFormEditContainer extends Component {
  constructor (props) {
    super(props)
    this.state = {id: this.props.match.params.id}
  }
  render () {
    let taskExists = this.props.tasks.find(task => task.id == this.state.id)
    if (taskExists) {
      return (
        <TaskFormComponent edit task={taskExists} editTask={this.props.editTask} />
      )
    } else {
      return (<Redirect to="/404" />)
    }
  }
}

function mapStateToProps (state) {
  return {tasks: state.tasks}
}
export default connect(mapStateToProps, { editTask })(TaskFormEditContainer)
