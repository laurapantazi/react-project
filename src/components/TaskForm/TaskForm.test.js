import {TaskForm} from './TaskForm'
import React from 'react'
import renderer from 'react-test-renderer'

it('TaskForm render correctly', () => {
  const tree = renderer.create(<TaskForm add addTask={() => {}} />).toJSON()
  expect(tree).toMatchSnapshot()
})