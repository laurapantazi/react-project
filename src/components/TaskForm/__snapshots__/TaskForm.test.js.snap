// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm render correctly 1`] = `
<div
  className="container taskList"
>
  <ul
    className="list-group"
    onKeyDown={[Function]}
  />
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formPlaintextDescription"
      >
        Description
      </label>
      <input
        className="form-control"
        id="formPlaintextDescription"
        name="description"
        onChange={[Function]}
        placeholder="Type description"
        type="text"
        value=""
      />
      <div
        className="valid-feedback"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formPlaintextCategory"
      >
        Category
      </label>
      <input
        className="form-control"
        id="formPlaintextCategory"
        name="category"
        onChange={[Function]}
        placeholder="Type category"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-row"
    >
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="formGridDuration"
        >
          Duration
        </label>
        <div
          className="input-group"
        >
          <input
            className="form-control"
            id="formGridDuration"
            name="duration"
            onChange={[Function]}
            placeholder="Type duration"
            type="number"
            value=""
          />
          <div
            className="input-group-append"
          >
            <span
              className="input-group-text"
              id="inputGroupPrepend"
            >
              minutes
            </span>
          </div>
        </div>
      </div>
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="formGridPriority"
        >
          Priority
        </label>
        <select
          className="form-control"
          id="formGridPriority"
          name="priority"
          onChange={[Function]}
          value=""
        >
          <option>
            Choose Priority
          </option>
          <option>
            High
          </option>
          <option>
            Medium
          </option>
          <option>
            Low
          </option>
        </select>
      </div>
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="formGridPoints"
        >
          Points
        </label>
        <input
          className="form-control"
          id="formGridPoints"
          name="points"
          onChange={[Function]}
          placeholder="Type points"
          step="5"
          type="number"
          value=""
        />
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
      value="ok"
    >
      Create
    </button>
  </form>
</div>
`;

exports[`TaskForm render correctly 2`] = `
<div
  className="container taskList"
>
  <ul
    className="list-group"
    onKeyDown={[Function]}
  />
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formPlaintextDescription"
      >
        Description
      </label>
      <input
        className="form-control"
        id="formPlaintextDescription"
        name="description"
        onChange={[Function]}
        placeholder="Type description"
        type="text"
      />
      <div
        className="valid-feedback"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formPlaintextCategory"
      >
        Category
      </label>
      <input
        className="form-control"
        id="formPlaintextCategory"
        name="category"
        onChange={[Function]}
        placeholder="Type category"
        type="text"
      />
    </div>
    <div
      className="form-row"
    >
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="formGridDuration"
        >
          Duration
        </label>
        <div
          className="input-group"
        >
          <input
            className="form-control"
            id="formGridDuration"
            name="duration"
            onChange={[Function]}
            placeholder="Type duration"
            type="number"
          />
          <div
            className="input-group-append"
          >
            <span
              className="input-group-text"
              id="inputGroupPrepend"
            >
              minutes
            </span>
          </div>
        </div>
      </div>
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="formGridPriority"
        >
          Priority
        </label>
        <select
          className="form-control"
          id="formGridPriority"
          name="priority"
          onChange={[Function]}
        >
          <option>
            Choose Priority
          </option>
          <option>
            High
          </option>
          <option>
            Medium
          </option>
          <option>
            Low
          </option>
        </select>
      </div>
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="formGridPoints"
        >
          Points
        </label>
        <input
          className="form-control"
          id="formGridPoints"
          name="points"
          onChange={[Function]}
          placeholder="Type points"
          step="5"
          type="number"
        />
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
      value="ok"
    >
      Save Changes
    </button>
  </form>
</div>
`;
